app/modules/ads/index.js

```diff
@@ -1,607 +1,71 @@
-
-    pageTracker.trackProp('session_viewability_all', viewabilityCounter.getViewability());
-    pageTracker.trackProp('session_viewability_tb', viewabilityCounter.getViewability('top_boxad'));
-    pageTracker.trackProp('session_viewability_icb', viewabilityCounter.getViewability('incontent_boxad'));
-
-    viewabilityCounter.init();
-  }
-
-  /**
-   * @private
-   */
-  trackLabradorToDW() {
-    const { utils, InstantConfigCacheStorage } = window.Wikia.adEngine;
-    const cacheStorage = InstantConfigCacheStorage.make();
-    const labradorPropValue = cacheStorage.getSamplingResults().join(';');
-
-    if (labradorPropValue) {
-      pageTracker.trackProp('labrador', labradorPropValue);
-      utils.logger(logGroup, 'labrador props', labradorPropValue);
-    }
-  }
-
   /**
-   * @private
+   * @type {PromiseLock}
    */
-  trackSpaInstanceId() {
-    const { context } = window.Wikia.adEngine;
+  static isExperimental;
 
-    if (!context.get('options.tracking.spaInstanceId')) {
+  static ensureMode(adsContext) {
+    if (Ads.isExperimental.isLoaded) {
       return;
     }
 
-    if (!this.spaInstanceId) {
-      this.spaInstanceId = uuid();
+    switch (getIsAdEngineExperimental()) {
+      case '1':
+        return Ads.isExperimental.resolve(true);
+      case '0':
+        return Ads.isExperimental.resolve(false);
+      default:
+        return Ads.isExperimental.resolve(!!(adsContext.opts && adsContext.opts.adEngineExperimental));
     }
-
-    pageTracker.trackProp('spa_instance_id', this.spaInstanceId);
   }
 
   /**
-   * @private
+   * Returns ads instance.
+   * @returns {OldAds | NewAds}
    */
-  trackTabId() {
-    const { context } = window.Wikia.adEngine;
-
-    if (!context.get('options.tracking.tabId')) {
-      return;
+  static getInstance() {
+    if (!Ads.isExperimental.isLoaded) {
+      console.error('ensureMode should be called before getInstance');
+      Ads.ensureMode({});
     }
 
-    window.tabId = sessionStorage.tab_id ? sessionStorage.tab_id : sessionStorage.tab_id = uuid();
-
-    pageTracker.trackProp('tab_id', window.tabId);
-  }
-
-  /**
-   * @private
-   */
-  trackConnectionToDW() {
-    const { utils } = window.Wikia.adEngine;
-    const connection = navigator.connection
-      || navigator.mozConnection
-      || navigator.webkitConnection;
-
-    if (connection) {
-      const data = [];
-      if (connection.downlink) {
-        data.push(`downlink=${connection.downlink.toFixed(1)}`);
-      }
-      if (connection.effectiveType) {
-        data.push(`effectiveType=${connection.effectiveType}`);
-      }
-      if (connection.rtt) {
-        data.push(`rtt=${connection.rtt.toFixed(0)}`);
-      }
-      if (typeof connection.saveData === 'boolean') {
-        data.push(`saveData=${+connection.saveData}`);
-      }
-
-      pageTracker.trackProp('connection', data.join(';'));
-      utils.logger(logGroup, 'connection', data);
+    if (Ads.isExperimental) {
```

Comment:
This is always truthy, am I wrong? Shouldn't it be?
```suggestion
    if (Ads.isExperimental.value) {
```